legate/driver/launcher.py

```diff
@@ -380,7 +389,7 @@ class SRunLauncher(Launcher):
     def __init__(self, config: ConfigProtocol, system: System) -> None:
         super().__init__(config, system)
 
-        self.rank_id = "%q{SLURM_PROCID}"
+        self.rank_id = LEGATE_GLOBAL_RANK_SUBSTITUTION
```

Comment:
Is there a reason to keep the old `.rank_id` field at this point? It looks like we could simply use `LEGATE_GLOBAL_RANK_SUBSTITUTION` directly in `command.py`.

Reply:
The thought occurred, I was just trying to keep changed minimal here. But it would allow deleting quite a bit of test complexity so I will proceed with this idea in a follow-on PR.

Reply:
I went ahead and did this here in https://github.com/nv-legate/legate.core/pull/471/commits/718fea851dca78824136e837efa4559b7f5c4756